File: TheAnvil/src/main/java/com/qualityplus/anvil/api/provider/EnchantmentProvider.java
package com.qualityplus.anvil.api.provider;

import com.qualityplus.anvil.api.session.AnvilSession.SessionResult;
import com.qualityplus.anvil.api.session.AnvilSession;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

public interface EnchantmentProvider {
    public SessionResult getAnswer(AnvilSession session);

    public ItemStack getFinalItem(AnvilSession anvilSession);

    public double getMoneyCost(AnvilSession session);

    public double getLevelsCost(AnvilSession session);

    public boolean dontHavePermission(AnvilSession session, Player player);
}

File: TheAnvil/src/main/java/com/qualityplus/anvil/api/provider/EnchantmentProvider.kt
package com.qualityplus.anvil.api.provider

import com.qualityplus.anvil.api.session.AnvilSession
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack

interface EnchantmentProvider {
    public fun getAnswer(session: AnvilSession): AnvilSession.SessionResult

    public fun getFinalItem(anvilSession: AnvilSession): ItemStack

    public fun getMoneyCost(session: AnvilSession): Double

    public fun getLevelsCost(session: AnvilSession): Double

    public fun dontHavePermission(session: AnvilSession, player: Player?): Boolean
}